starport/pkg/goanalysis/goanalysis.go

```diff
@@ -0,0 +1,53 @@
+package goanalysis
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/fs"
+)
+
+type Type struct {
+	Name string
+}
+
+func FindTypes(srcDirPath string, names []string) ([]Type, error) {
```

Comment:
Hey @bensooraj, thanks for the contribution!

This func basically needs to find out Go types in the source code by the given `names` args. But with this func signature we cannot be sure that returned `Type`s slice actually contains all the `names` requested. To find out not found types, caller needs to make an additional check. I beleieve this func signature is a bit spooky and can be simplified with something like below:

```go
FindTypes(src string, walker func(Type) (continue bool)) error
```

In this version, `walker` should be called on each type we find in `src` and caller itself makes the filtering based on its custom checks. This is actually a bit more flexible. Also, I added a `continue` return value to the `walker` to indicate if finding process should continue or stop.